rust/connlib/tunnel/src/client.rs

```diff
@@ -950,6 +858,100 @@ impl ClientState {
     pub(crate) fn poll_transmit(&mut self) -> Option<snownet::Transmit<'_>> {
         self.node.poll_transmit()
     }
+
+    pub fn set_resources(&mut self, resources: &[ResourceDescription]) {
```

Comment:
It feels a bit odd to pass a reference here. I'd expect a "setter" to be given an owned value.